cnd/src/http_api/swap_resource.rs

```diff
@@ -168,10 +182,14 @@ pub fn build_rfc003_siren_entity<S: StateStore>(
                 .with_type("text/html")
                 .with_class_member("protocol-spec"),
             );
-        let entity = actions.into_iter().fold(entity, |acc, action| {
-            let action = action.to_siren_action(&id);
-            acc.with_action(action)
-        });
+
+        if on_fail == OnFail::NoAction {
```

Comment:
Hum, just do a `match` block? :)

Reply:
This code is buggy, is now:
```
if state.swap_failed() && on_fail == OnFail::NoAction {
```

Reply:
So actually a match block within `if state.swap_failed()` would help clarify the logic. (ok looking at the code now).